pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/url_test.go

```diff
@@ -0,0 +1,119 @@
+package ingress
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+)
+
+func TestGenerateUrl(t *testing.T) {
+	type args struct {
+		name          string
+		namespace     string
+		ingressConfig *v1beta1.IngressConfig
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "empty url template",
+			args: args{
+				name:      "model",
+				namespace: "user",
+				ingressConfig: &v1beta1.IngressConfig{
+					IngressDomain: "my.domain",
+				},
+			},
+			want: "",
+		},
+		{
+			name: "path based url template",
+			args: args{
+				name:      "model",
```

Comment:
is this the service name or model name? There could be multiple models in a given InferenceService

Reply:
This is service (isvc) name. 
> There could be multiple models in a given InferenceService

ACK. I believe the value here, `model`, was taken from the existing `domain_test.go`. But if you want we can change it to something else.